### PR TITLE
feat: add missing project warning

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/warning/warning_key_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import (
+	"github.com/ovh/cds/engine/api/application"
+	"github.com/ovh/cds/engine/api/bootstrap"
+	"github.com/ovh/cds/engine/api/pipeline"
+	"github.com/ovh/cds/engine/api/project"
+	"github.com/ovh/cds/engine/api/test"
+	"github.com/ovh/cds/engine/api/test/assets"
+	"github.com/ovh/cds/sdk"
+	"github.com/stretchr/testify/assert"
+	"testing"
```
Comment (yesnault) [inline engine/api/warning/warning_key_test.go]: imp

File: engine/api/warning_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"encoding/json"
```
Comment (yesnault) [inline engine/api/warning_test.go]: import

File: engine/api/application/application_vcs.go [hunk 3]
```diff
@@ -36,3 +38,35 @@ func DecryptVCSStrategyPassword(app *sdk.Application) error {
 	app.RepositoryStrategy.Password = string(clearPWD)
 	return nil
 }
+
+// CountKeysInVcsConfiguration counts key use in application vcs configuration for the given project
+func CountKeysInVcsConfiguration(db gorp.SqlExecutor, projectKey string, vcsName string) ([]string, error) {
```
Comment (fsamin) [inline engine/api/application/application_vcs.go]: rename

File: engine/api/environment/environment.go [hunk 4]
```diff
@@ -473,37 +473,21 @@ func Permission(key string, envName string, u *sdk.User) int {
...
 
-// CountInValueVarData represents the result of CountInVarValue function
-type CountInValueVarData struct {
-	Name  string
-	Count int64
-}
-
 // CountInVarValue counts how many time a pattern is in variable value for the given project
-func CountInVarValue(db gorp.SqlExecutor, key string, value string) ([]CountInValueVarData, error) {
+func CountInVarValue(db gorp.SqlExecutor, projectKey string, value string) ([]string, error) {
```
Comment (fsamin) [inline engine/api/environment/environment.go]: rename this function

Comment (fsamin) [reply engine/api/environment/environment.go]: i can't see any test for this ?

File: engine/api/warning/warning.go [hunk 5]
```diff
@@ -61,3 +62,21 @@ func commit(tx *gorp.Transaction) {
 	}
 	return
 }
+
+func pipelineSliceMerge(paramDataSlice []pipeline.CountInValueParamData, pipDataSlice []pipeline.CountInPipelineData) []string {
+	elementsMap := make(map[string]bool)
+	pipelinesName := make([]string, 0)
```
Comment (fsamin) [inline engine/api/warning/warning.go]: allocation

File: engine/api/warning.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	"net/http"
+	"text/template"
+
+	"github.com/gorilla/mux"
+
+	"github.com/ovh/cds/engine/api/warning"
+	"github.com/ovh/cds/sdk"
+)
+
+func (api *API) getWarningsHandler() Handler {
```
Comment (fsamin) [inline engine/api/warning.go]: i miss the sdk part of it, and probably the warning display in cdsctl (can be done later)

**Pull request merged**